src/scripts/index.js

```diff
@@ -0,0 +1,302 @@
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "e775eb01-ae86-4e68-acff-d832598a5ddf",
+    "Content-Type": "application/json",
+  },
+});
+
+const profileEditButton = document.querySelector(".profile__edit-btn");
+const cardModalBtn = document.querySelector(".profile__add-btn");
+const profileName = document.querySelector(".profile__name");
+const profileDescription = document.querySelector(".profile__description");
+const avatarModalBtn = document.querySelector(".profile__avatar-btn");
+const profileAvatar = document.querySelector(".profile__avatar");
+
+const editModal = document.querySelector("#edit-modal");
+const editFormElement = document.forms["edit-profile"];
+const editModalClosedBtn = editModal.querySelector(".modal__close-btn");
+const editModalNameInput = editModal.querySelector("#profile-name-input");
+const editModalDescriptionInput = editModal.querySelector(
+  "#profile-description-input"
+);
+
+const cardModal = document.querySelector("#add-card-modal");
+const cardForm = document.forms["new-post"];
+const cardSubmitBtn = cardModal.querySelector(".modal__button");
+const cardModalClosedBtn = cardModal.querySelector(".modal__close-btn");
+const cardNameInput = cardModal.querySelector("#add-card-name-input");
+const cardLinkInput = cardModal.querySelector("#add-card-link-input");
+
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+const previewModalCloseBtn = previewModal.querySelector(".modal__close-btn");
+
+//AVATAR form element
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarSubmitBtn = avatarModal.querySelector(".modal__button");
+const avatarModalCloseBtn = avatarModal.querySelector(".modal__close-btn");
+const avatarInput = avatarModal.querySelector("#profile-avatar-input");
+
+let selectedCard;
+let selectedCardID;
+
+//Deete form elements
+const deleteModal = document.querySelector("#delete-modal");
+const cancelDeleteBtn = deleteModal.querySelector(
+  ".modal__submit-btn[type='button']"
+);
+const deleteForm = deleteModal.querySelector(".modal__form");
+
+const allModals = document.querySelectorAll(".modal");
+
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+const deleteIcon = document.querySelector(".close_delete-icon");
+
+//TODO -Destructure the secon item in the callback of the ,then()
+api
+  .getAppInfo()
+  .then(([user, cards]) => {
+    console.log(cards);
+    profileName.textContent = user.name;
+    profileDescription.textContent = user.about;
+    profileAvatar.src = user.avatar;
+
+    cards.forEach((card) => {
+      const cardEl = getCardElement(card);
+      cardsList.append(cardEl);
```

Comment:
**Could be improved**

You can make a universal function for adding a card into the section using different methods: `append`, `prepend`, or anything else:

```js

// The function accepts a card object and a method of adding to the section
// The method is initially `prepend`, but you can pass `append` 
function renderCard(item, method = "prepend") {
  
  const cardElement = getCardElement(item);
  // Add the card into the section using the method
  cardsList[ method ](cardElement);
}



```

Every `DOM` element is an **object** with methods, that’s why you can get the methods in 2 ways: `cardsList.prepend(…)` (with a **dot**) or `cardsList["prepend"](…)`. It’s the same.